src/aframe-streetmix-parsers.js

```diff
@@ -151,6 +151,49 @@ function createClonedVariants (variantName, positionX, clonedObjectRadius, step
   return dividerParentEl;
 }
 
+function getRandomIntInclusive(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min + 1) + min);
+}
+
+function getRandomArbitrary(min, max) {
+  return Math.random() * (max - min) + min;
+}
+
+function getZPositions(start, end, step) {
+  const len = Math.floor((end - start) / step) + 1
+  var arr = Array(len).fill().map((_, idx) => start + (idx * step))
+  return arr.sort( () => .5 - Math.random() );
+}
+
+function createSidewalkClonedVariants (BasePositionX, segmentWidthInMeters, density) {
+  var xValueRange = [-(.37*segmentWidthInMeters), (.37*segmentWidthInMeters)];
+  var zValueRange = getZPositions(-32,32,1.5);
+  var totalPedestrianNumber;
+  if (density === "sparse"){
+    totalPedestrianNumber = 5.0;
+  } else if (density === "normal") {
+    totalPedestrianNumber = 10.0;
+  } else {
+    totalPedestrianNumber = 15.0;
+  }
+  const dividerParentEl = createParentElement(BasePositionX, `${variantName}-parent`);
+  //Randomly generate avatars
+  for (let i = 0; i < totalPedestrianNumber; i++){
+    var variantName = "char" + String(getRandomIntInclusive(1,16));
+    var positionXYZString = getRandomArbitrary(xValueRange[0], xValueRange[1]) + ' 0 ' + zValueRange.pop();
+    var placedObjectEl = document.createElement('a-entity');
+    placedObjectEl.setAttribute('class', variantName);
```

Comment:
we don't really need class name, it's redundant to mixin and we should prbly remove it in other places too

Reply:
Understood :+1: I just added a commit that removes line 187:
`placedObjectEl.setAttribute('class', variantName);`
I am happy to remove the class name from other places in `aframe-streetmix-parsers.js`, but that is probably beyond the scope of this pull request.